packages/venia-concept/src/components/MiniCart/miniCart.js

```diff
@@ -88,7 +90,9 @@ class MiniCart extends Component {
                         <Icon name="x" />
                     </Trigger>
                 </div>
-                {cart.details.items && cart.details.items.length ? (
+                {isCartEmpty ? (
+                    <EmptyMiniCart />
```

Comment:
Avoid multiline expressions in JSX; it breaks the ability to read the tree. Consider returning this expression from a [helper method](https://github.com/magento-research/pwa-studio/blob/release/2.0/packages/peregrine/src/Router/MagentoRouteHandler.js#L104-L112) or a [getter](https://github.com/magento-research/pwa-studio/blob/release/2.0/packages/venia-concept/src/components/Checkout/flow.js#L43), or [defining it as a variable outside of the return value and rendering the variable](https://github.com/magento-research/pwa-studio/blob/release/2.0/packages/peregrine/src/List/items.js#L61-L78).